src/wsi/wsi_platform.cpp

```diff
@@ -0,0 +1,169 @@
+#include "wsi_platform.h"
+#include "wsi_monitor.h"
+#include "wsi_window.h"
+#include "../util/util_env.h"
+#include "../util/util_error.h"
+
+namespace dxvk::wsi {
+  static WsiDriver* s_driver = nullptr;
+  static int s_refcount = 0;
+
+  static const WsiBootstrap *wsiBootstrap[] = {
+#if defined(DXVK_WSI_WIN32)
+    &Win32WSI,
+#endif
+#if defined(DXVK_WSI_SDL2)
+    &Sdl2WSI,
+#endif
+#if defined(DXVK_WSI_GLFW)
+    &GlfwWSI,
+#endif
+  };
+
+  void init() {
+    if (s_refcount++ > 0)
+      return;
+
+    bool success = false;
+    const std::string hint = dxvk::env::getEnvVar("DXVK_WSIDRIVER");
+    for (const WsiBootstrap *b : wsiBootstrap) {
+      if ((hint != "") && (hint != b->name))
+          continue;
```

Comment:
Talking to @Joshua-Ashton, I think a potential issue here is that if multiple WSIs are available we might initialize a random one that the app might not be able to deal with if the env var isn't set. Maybe we should outright fail if we have no env var, or just default to the first WSI in the list?

Reply:
I'd be okay with that; it's something I've run into particularly with SDL3 migration - will make this change later this morning!